source/materials.c

```diff
@@ -109,7 +114,7 @@ int aws_cryptosdk_keyring_on_encrypt(
     /* Postcondition: If this keyring generated data key, it must be the right length. */
     if (!precall_data_key_buf.buffer && unencrypted_data_key->buffer) {
         const struct aws_cryptosdk_alg_properties *props = aws_cryptosdk_alg_props(alg);
-        if (unencrypted_data_key->len != props->data_key_len) {
+        if (props == NULL || (unencrypted_data_key->len != props->data_key_len)) {
```

Comment:
@ttjsu-aws in case `props == NULL`, we should also return error, do you confirm?